Libplanet/Net/Protocols/KademliaProtocol.cs

```diff
@@ -13,38 +13,38 @@ namespace Libplanet.Net.Protocols
     internal class KademliaProtocol<T> : IProtocol
         where T : IAction, new()
     {
-        private const int BucketSize = 16;
-        private const int TableSize = Address.Size * sizeof(byte) * 8;
-        private const int FindConcurrency = 3;
-        private const int MaxDepth = 3;
-
-        // FIXME: This should be configurable?
-        private static readonly TimeSpan RequestTimeout = TimeSpan.FromSeconds(5);
-        private static readonly TimeSpan IdleRefreshInterval = TimeSpan.FromSeconds(5);
-        private static readonly TimeSpan IdleRebuildInterval = TimeSpan.FromMinutes(30);
-
+        private readonly TimeSpan _requestTimeout;
         private readonly Swarm<T> _swarm;
         private readonly Address _address;
         private readonly int _appProtocolVersion;
         private readonly System.Random _random;
         private readonly RoutingTable _routing;
+        private readonly int _tableSize;
+        private readonly int _bucketSize;
 
         private readonly ILogger _logger;
 
         public KademliaProtocol(
             Swarm<T> swarm,
             Address address,
             int appProtocolVersion,
-            ILogger logger)
+            ILogger logger,
+            int? tableSize,
+            int? bucketSize,
```

Comment:
How about giving it a default value instead of allowing it to be `null`?

```suggestion
            int bucketSize = Kademlia.BucketSize,
```

Reply:
If default value is given, cyclic constructor call will happen since public method without those property is exactly same with internal method.

It can be one way to just expose those parameters as public I guess...